### PR TITLE
Parent Email Banner

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Screenshot 2020-04-15 at 1 00 45 PM

## Links

- [adhoc](https://adhoc-parent-email-banner-studio.cdn-code.org/)
- [spec](https://docs.google.com/document/d/1iOfV0Opkg9Jqw3LEmKtmq4KCWbmeCnTkSYvf-_BNYIg/edit#heading=h.6jx150u3fizm)
- [jira](https://codedotorg.atlassian.net/browse/FND-1101)

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by daynew** [comment]
minor nit:
content isn't centered vertically
image

**Comment by daynew** [comment]
Have you asked about the choice of the color puple for the background? The banner is not noticeable when it overlaps the footer.
image

**Comment by bethanyaconnor** [comment]
> minor nit:
> content isn't centered vertically
> image

Which element are you referring to? It's not clear to me from the screenshot

**Comment by bethanyaconnor** [comment]
> Have you asked about the choice of the color puple for the background? The banner is not noticeable when it overlaps the footer.
> image

That's what it looks like in the spec and it's the same as the cookie banner. @dju90 any thoughts?

File: dashboard/app/views/layouts/_parent_email_banner.haml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+-# Only show when we're not showing the cookie banner,
+-# the current user is signed in, a student, doesn't have a parent email, and hasn't previously dismissed the banner
+-# and the page isn't an Applab or Gamelab level.
+
+:ruby
+  show_banner =
+    !(request.gdpr? || (rack_env == :test && request.params['show_cookie_banner_on_test'].present?)) &&
```
Comment (daynew) [inline dashboard/app/views/layouts/_parent_email_banner.haml:7]: Why is the banner hidden for GDPR countries?

Comment (bethanyaconnor) [reply dashboard/app/views/layouts/_parent_email_banner.haml:7]: Because we show the cookie banner for those users

**Comment by daynew** [comment]
Is it intentional that the button styles changes when I load Minecraft Designer? Not sure if bug of feature :)
image

**Comment by daynew** [comment]
> > minor nit:
> > content isn't centered vertically
> > image
> 
> Which element are you referring to? It's not clear to me from the screenshot

I highlighted in red the unequal gaps. All the elements (text and buttons) are not centered vertically.

File: dashboard/app/views/layouts/_parent_email_banner.haml (hunk 1 above)
Comment (daynew) [reply dashboard/app/views/layouts/_parent_email_banner.haml:7]: My bad. I thought the cookie banner might be like this one: it doesn't show up if the user has accepted it already. If it wasn't isn't shown, then we could show this one...

I've never seen the cookie banner and don't know the business logic.

**Review by daynew — APPROVED** [review]
Overall looks good. I poked around on the adhoc and left some minor style comments.

**Comment by dju90** [comment]
> Is it intentional that the button styles changes when I load Minecraft Designer? Not sure if bug of feature :)
> image

Would ideally like to keep the button style the same as on the homepage, even in minecraft designer. But can also choose to fix in later iteration

> minor nit:
> content isn't centered vertically
> image

If it's an easy fix, sure; otherwise, fine to leave as is

> > Have you asked about the choice of the color puple for the background? The banner is not noticeable when it overlaps the footer.
> > image
> 
> That's what it looks like in the spec and it's the same as the cookie banner. @dju90 any thoughts?

Let's keep the color; I'm not too concerned about users scrolling to the bottom and not noticing - they'll soon navigate away and see it with better contrast

**Comment by bethanyaconnor** [comment]
> Is it intentional that the button styles changes when I load Minecraft Designer? Not sure if bug of feature :)
> image

Good catch! Not intentional -- I'll fix that!

**Comment by bethanyaconnor** [comment]
I believe the Minecraft issue is an easy fix and an artifact of me trying to clean up some CSS that I copied. 🙃 Working on that now.

I'll poke at the vertical alignment and see if I can get it to align a bit better but I won't spend too long on it.

👍 on the colors

**Pull request merged**